### DISCUSSION TITLE
Does S2 support vertical top and bottom row headers?

### DISCUSSION THREAD

**Discussion by heydong1** [antvis/S2#2040] [Q&A] [answered]
The default is vertical centering, I don't see the relevant configuration, please advise, thank you very much!

**Comment by lijinke666** [comment]
Not supported at the moment, can you tell me the usage scenario?
Now. only support textAlign https://s2.antv.antgroup.com/api/general/s2theme#texttheme

**Comment by heydong1** [comment, marked as answer]
This is the experience optimization requirements proposed by the designer, and also some of the customers. Fortunately, I looked at the source code and rewrote the `getAdjustPosition` method to implement this requirement. Here is [observable notebook](https://observablehq.com/d/7b5c43323fdab1e0). Just change the boundary of the text and add some paddingY, perhaps I need to consider the baseline.
